platform/linux-generic/odp_schedule_if.c

```diff
@@ -51,18 +48,20 @@ int odp_schedule_config(const odp_schedule_config_t *config)
 	int ret;
 	odp_schedule_config_t defconfig;
 
-	ODP_ASSERT(!_odp_schedule_configured);
+	if (_odp_schedule_configured) {
+		ODP_ERR("Scheduler has been configured already\n");
+		return -1;
```

Comment:
I think here should not be return. 2 odp_schedule_config() should be valid for user or unconfig() function has to be provided.

Reply:
There's no point of application calling config() multiple times as only the last config call would apply. Single config() call enable implementation to configure and start HW scheduler at this point. Typically, HW scheduler cannot be reconfigured without reboot.

I'll add note to the API spec that config() can be called only once.

Reply:
Config() can be called only once per instance (once after global_init()). API does not require synchronization between instances. The point is that this is not supported:

global_init()
...
config()
config()
config()
...
schedule()

Reply:
@muvarov if you have such an applicaiton, it will call `capability()`, select optimum values and then call `config()`. Other subsystems (IPsec) follow same approach - you can call `config()` only once.